### PR TITLE
update with cmake 3.21 and actions

### PR DESCRIPTION
**Pull request by daveb-dev**

### CODE REVIEW

File: FMCA/src/util/Macros.h [hunk 1]
```diff
@@ -29,7 +29,8 @@ namespace FMCA {
 #define FMCA_ZERO_TOLERANCE std::numeric_limits<FMCA_SCALAR>::epsilon()
 #define FMCA_BBOX_THREASHOLD FMCA_ZERO_TOLERANCE
 
-typedef FMCA_INDEX Index;
```
Comment (muchip) [inline FMCA/src/util/Macros.h:32]: Please keeep FMCA_INDEX here. This is safer as this is used everywhere else

**Review by muchip — CHANGES_REQUESTED** [review]
What changed in the jupyter notebooks? Please fix the Macros.h

File: FMCA/src/util/Macros.h (hunk 1 above)
Comment (daveb-dev) [reply FMCA/src/util/Macros.h:32]: > What changed in the jupyter notebooks? Please fix the Macros.h

Markdown documentation of how to apply pseudo Inverse

**Comment by daveb-dev** [comment]
New pull request will address these feedback and rebase with master

**Pull request closed without merging**